spinalcordtoolbox/deepseg/inference.py

```diff
@@ -77,3 +91,158 @@ def segment_and_average_volumes(model_paths, input_filenames, options):
     target_lst = target_lsts[0]
 
     return nii_lst, target_lst
+
+
+def segment_non_ivadomed(path_model, model_type, input_filenames, threshold):
+    # MONAI and NNUnet have similar structure, and so we use nnunet+inference functions with the same signature
+    if model_type == "monai":
+        create_net = ds_monai.create_nnunet_from_plans
+        inference = segment_monai
+    else:
+        assert model_type == "nnunet"
+        create_net = ds_nnunet.create_nnunet_from_plans
+        inference = segment_nnunet
+
+    # equivalent to `with torch.no_grad()`
+    torch.set_grad_enabled(False)
+
+    # load model from checkpoint
+    net = create_net(path_model)
+
+    nii_lst, target_lst = [], []
+    for fname_in in input_filenames:
+        # model may be multiclass, so the `inference` func should output a list of fnames and targets
+        fnames_out, targets = inference(path_img=fname_in, tmpdir=tmp_create(basename="sct_deepseg"), predictor=net)
+        for fname_out, target in zip(fnames_out, targets):
+            # TODO: Use API binarization function when output filetype is sct.image.Image
+            run_proc(["sct_maths", "-i", fname_out, "-bin", str(threshold), "-o", fname_out])
+            # TODO: Change the output filetype from Nifti1Image to sct.image.Image to mitigate #3232
+            nii_lst.append(nib.load(fname_out))
+            target_lst.append(target)
+
+    return nii_lst, target_lst
+
+
+def segment_monai(path_img, tmpdir, predictor):
+    """
+    Script to run inference on a MONAI-based model for contrast-agnostic soft segmentation of the spinal cord.
+
+    Author: Naga Karthik
+    Original script: https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/e65478099d026f865b7f1d7d0082e6e9a507a744/monai/run_inference_single_image.py
+    """
+    # Copy the file to the temporary directory using shutil.copyfile
+    path_img_tmp = os.path.join(tmpdir, os.path.basename(path_img))
+    shutil.copyfile(path_img, path_img_tmp)
+    logger.info(f'Copied {path_img} to {path_img_tmp}')
+
+    # define inference patch size and center crop size
+    crop_size = (64, 192, -1)
```

Comment:
do we want the option to change this?

Reply:
yes, this should be an input arg when using `sct_deepseg`!

Reply:
> different input data requires different crops

Precisely! We are providing it is an input arg in the original inference file because we would want the users to decide how to crop based on the type of images they (i.e. ones with small/large field-of-view, more/less spinal curvature, etc.)

> something that is model-dependent

In fact, what is model-depended is `inference_roi_size` which has to be fixed (and is rightly done in the current code as `64x192x320`)

> (Do they examine their input image in a 3D viewer and estimate the center region containing the spinal cord? 

Yes, this is what we expect. 

> How much of it would be trial and error?)

Although, the default suggestion should work fine, we thought that having the user play with the crop sizes would be useful for them. This is something we also discussed in this [issue](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/issues/70) but Idk how to move forward with Julien's suggestion in that issue.

Reply:
Ahhh, thank you for linking the issue. The background context helps a lot! It looks like you and @jcohenadad had the same thoughts, hehehe:

> But, the problem is it is confusing from a user standpoint to know how to crop. Details in https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/pull/60#discussion_r1338812249

Reply:
As per the 2024-01-24 meeting, we will follow up in a separate PR, as to not hold up the initial PR.

Issue created --> https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4349